src/apps/replication/lib/replica_2pc.cpp

```diff
@@ -364,6 +377,8 @@ void replica::on_append_log_completed(mutation_ptr& mu, error_code err, size_t s
         }
 
         // mutation log failure, propagted to all replicas
+        // TODO(qinzuoyan): what if log failed for old mutations? it will not reach here,
```

Comment:
you are right, this should be moved out of the if-condition before to handle all log failures. Since the chance of log failure small, let's fail fast to make failure handling simple.